### PR TITLE
Clean up the repository

### PR DESCRIPTION
**Pull request by powerduncan**
Lots of cleanup changes across the entire repository:

* Simplify the `Makefile` setup (only a single `Makefile`, with pattern rules to automatically generate Go and Python sources for Protocol Buffers).
* Check in generated Protocol Buffer sources so that developers only need `protoc` if they are changing a message or service.
* Add Python package configuration to support namespaced imports (`import kinney.orchestrator.api`), both in a real installation (*e.g.*, `pip install kinney`) and in development (*e.g.*, `pipenv run python -m kinney.orchestrator.server`).
* Remove the unnecessary `ev/` directory.
* Move the ChargePoint Python code into a `py/` subdirectory.
* Rename the directory containing the Go ChargePoint API code to the more appropriate `api/`.
* Configure YAPF for Python formatting, and apply it to all Python files in the repository.
* Remove the unnecessary gRPC/REST proxy and associated dependencies.
* Style edits to the main `README.md` file, and added a brief section on setting up the development environment.
* Move installation of the Go Protocol Buffer compiler plugins to use `go build` with a specific local destination, rather than relying on `$GOPATH` and `go get`.
* Add explicit dependencies on the Go modules that provide the necessary Protocol Buffer compiler plugins (see `util/tools.go`), so that they are included in `go.mod` and `go.sum`.
* Remove the unnecessary and editor-specific sections of the `.gitignore` file: developers should use custom local `.gitignore` files that apply to their own specific development environment and tools.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,6 +27,10 @@ bin/protoc-gen-go-grpc:
 %_grpc.pb.go: %.proto bin/protoc-gen-go-grpc
 	protoc --go-grpc_out="module=${GO_MODULE}:." --plugin="./bin/protoc-gen-go-grpc" "$<"
 
+# Helper rule to regenerate all Protocol Buffer sources at once.
+protos: orchestrator/api.pb.go orchestrator/api_grpc.pb.go
```
Comment (aaj-gd) [inline Makefile:104]: cant this be orchestrator/%.pb.go

File: Makefile [hunk 2]
```diff
@@ -26,24 +47,40 @@ bin/protoc-gen-go-grpc:
...
+# https://grpc.io/docs/languages/python/basics/#generating-client-and-server-code
+%_pb2_grpc.py: %.proto pipenv-dev
+	pipenv run python -m grpc_tools.protoc \
+		--proto_path="." \
+		--grpc_python_out="." \
+		"$<"
 
 # Helper rule to regenerate all Protocol Buffer sources at once.
 protos: orchestrator/api.pb.go orchestrator/api_grpc.pb.go
+protos: orchestrator/api_pb2.py orchestrator/api_pb2_grpc.py
```
Comment (aaj-gd) [inline Makefile:105]: shouldnt these lines be combined? does make like the same target twice.

File: Makefile [hunk 3]
```diff
@@ -26,24 +47,40 @@ bin/protoc-gen-go-grpc:
...
 .PHONY: protos
 
+################################################################################
+## Cleanup
+################################################################################
+
 clean:
-	rm -r ./bin/
+	-rm -r ./bin/
```
Comment (aaj-gd) [inline Makefile:114]: makefile black magic `-` means ignore exit condition. not sure if everyone will get that.

File: Makefile [hunk 4]
```diff
@@ -1,17 +1,31 @@
+# This Makefile is assumed to be in the root of the repository, so its location
+# is used to derive the directory that is the repository root.
```
Comment (aaj-gd) [inline Makefile:2]: not sure if this is any better but i usually just use git to tell me the root
`GIT_DIR="$(git rev-parse --show-toplevel)"`

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+# better cooperate with other languages in the same repository, there is no
+# "kinney" directory.  Instead, this behavior is simulated by the `package_dir`
+# directive indicating that the source directory for the "kinney" package is the
+# repository root (`package_dir={'kinney': ''}` in `setup.py`).
+#
+# Note, however, that this setup is *not* supported by "development mode" ([1]),
+# so that functionality is implemented by a "*.pth" file installed by
+# `util/install_pth.py`.
+#
+# [1]: https://setuptools.readthedocs.io/en/latest/setuptools.html#development-mode
```
Comment (aaj-gd) [inline setup.cfg:17]: wow this is impressive you prevented kinney/kinney

Comment (powerduncan) [reply setup.cfg:17]: thanks :) i'm rather proud of it myself!

**Pull request merged**